### PR TITLE
NEVISACCESSAPP-5323: Add GitHub actions

### PR DESCRIPTION
**Pull request by tamas-toth**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -689,3 +689,10 @@ hs_err_pid*
 !/gradle/wrapper/gradle-wrapper.jar
 
 # End of https://www.toptal.com/developers/gitignore/api/visualstudiocode,intellij,reactnative,xcode,node,macos,android,androidstudio,ruby,cocoapods,buck
```
Comment (daniel-toth-leeder) [inline .gitignore:691]: I'd regenerated the whole file with the tool used originally and update this list in the comment as well.

**Review by daniel-toth-leeder — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (balazs-gerlei) [reply .gitignore:691]: Agreed.

Comment (tamas-toth) [reply .gitignore]: Cool, I didn't know this tool, thanks for the suggestion, fixed.

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+[*]
+
+indent_style = tab
+indent_size = 4
+
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
```
Comment (balazs-gerlei) [inline .editorconfig:15]: Oh I wanted to bring this up for a while, thanks for adding it. Honestly, I think we should enable this for all of our IDEs for all of our projects...

File: .gitignore [hunk 3]
```diff
@@ -554,6 +576,11 @@ build-iPhoneSimulator/
 
 ### VisualStudioCode ###
 .vscode/*
+!.vscode/settings.json
+!.vscode/tasks.json
+!.vscode/launch.json
+!.vscode/extensions.json
+!.vscode/*.code-snippets
```
Comment (balazs-gerlei) [inline .gitignore:583]: I understand that this is part of the template but I think we don't need these excludes. E.g. I don't think it's necessary to commit my VS Code workspace settings for a public example app.

Comment (balazs-gerlei) [inline .gitignore:583]: I understand that this is part of the template but I think we don't need these excludes. E.g. I don't think it's necessary to commit my VS Code workspace settings for a public example app.

File: README.md [hunk 4]
```diff
@@ -65,7 +65,7 @@ First open a terminal and run the `yarn` command in the root directory to get th
 
 4. Synchronize your [android](/android) project with Gradle if opened in Android Studio.
 
-> **Warning**  
+> **Warning**
```
Comment (balazs-gerlei) [inline README.md:68]: Please update these to the current GitHub format (see [docs](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts)):
```markdown
> [!WARNING]
> ...
```

File: README.md [hunk 5]
```diff
@@ -173,7 +173,7 @@ yarn ios:old
 
 Or by choosing Product > Run from Xcode, or by clicking the Run button in your project’s toolbar, similarly how you would run a native iOS application.
 
-> **_NOTE_**  
+> **_NOTE_**
```
Comment (balazs-gerlei) [inline README.md:176]: Please update these to the current GitHub format (see [docs](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts)):
```markdown
> [!NOTE]
> ...
```

File: android/app/build.gradle [hunk 6]
```diff
@@ -93,6 +93,8 @@ dependencies {
     // The version of react-native is set by the React Native Gradle Plugin
     implementation("com.facebook.react:react-android")
 
+    implementation(platform("org.jetbrains.kotlin:kotlin-bom:1.8.0"))
```
Comment (balazs-gerlei) [inline android/app/build.gradle:96]: Why do we need this?

File: android/fastlane/Appfile [hunk 7]
```diff
@@ -0,0 +1,2 @@
+json_key_file("") # Path to the json secret file - Follow https://docs.fastlane.tools/actions/supply/#setup to get one
+package_name("ch.nevis.mobile.authentication.sdk.react.example") # e.g. com.krausefx.app
```
Comment (balazs-gerlei) [inline android/fastlane/Appfile:2]: I don't think we need the example in the comment.

**Review by balazs-gerlei — CHANGES_REQUESTED** [review]

File: .editorconfig (hunk 2 above)
Comment (tamas-toth) [reply .editorconfig:15]: Maybe we can bring it up on a backlog grooming...

File: .gitignore (hunk 3 above)
Comment (tamas-toth) [reply .gitignore:583]: Ok, those are deleted.

File: README.md (hunk 4 above)
Comment (tamas-toth) [reply README.md:68]: Thanks, fixed.

File: README.md (hunk 5 above)
Comment (tamas-toth) [reply README.md:176]: Fixed.

File: android/fastlane/Appfile (hunk 7 above)
Comment (tamas-toth) [reply android/fastlane/Appfile:2]: This is an autogenerated file by Fastlane, I don't really want to touch this...

File: android/app/build.gradle (hunk 6 above)
Comment (tamas-toth) [reply android/app/build.gradle:96]: It's an interesting thing. I got CI build errors (using old arch? - can't remember) regarding that different dependencies were compiled with different kotlin version. Adding this dependency resolved this issue, aligns all the Kotlin SDK libraries with the same version. You can read more [here](https://medium.com/@gabrielshanahan/a-deep-dive-into-an-initial-kotlin-build-gradle-kts-8950b81b214).

**Review by balazs-gerlei — APPROVED** [review]

**Review by daniel-toth-leeder — APPROVED** [review]

**Pull request merged**